### PR TITLE
fix compilation when imagemagick is v7

### PR DESCRIPTION
**Pull request by Izopi4a**
Hello!

when image magick version is 7 compilation fails 

https://github.com/mkoppanen/imagick/issues/180

### CODE REVIEW

File: phalcon/image/adapter/imagick.zep [hunk 1]
```diff
@@ -352,13 +352,22 @@ class Imagick extends Adapter
...
+		var watermark, ret, imagickVersion, method;
 
 		let opacity = opacity / 100,
-			watermark = new \Imagick();
+			watermark = new \Imagick(),
+			imagickVersion = \Imagick::getVersion();
+
+		preg_match("/ImageMagick ([0-9]+\.[0-9]+\.[0-9]+)/", imagickVersion["versionString"], imagickVersion);
+
+		if version_compare(imagickVersion[1], '7') < 0 {
```
Comment (sergeyklay) [inline phalcon/image/adapter/imagick.zep]: `'7'` -> "7"

**Review by sergeyklay — CHANGES_REQUESTED** [review]

**Comment by sergeyklay** [comment]
Also could you please update `CHANGELOG.md`

**Comment by elibyy** [comment]
why not use method_exists?

**Comment by Izopi4a** [comment]
well i might have to cuz preg_match might be broken in php5 ( for zephir ) , but if I use method exists , if they change that shit again in imagick 8 the code will look funny xD

**Comment by sergeyklay** [comment]
@Izopi4a Well, as you can see tests fails

File: phalcon/image/adapter/imagick.zep [hunk 2]
```diff
@@ -352,13 +352,22 @@ class Imagick extends Adapter
...
 	{
-		var watermark, ret;
+		var watermark, ret, imagickVersion, method, tmp;
 
 		let opacity = opacity / 100,
-			watermark = new \Imagick();
+			watermark = new \Imagick(),
+			imagickVersion = \Imagick::getVersion();
+
+		preg_match("/ImageMagick ([0-9]+\.[0-9]+\.[0-9]+)/", imagickVersion["versionString"], tmp);
```
Comment (Jurigag) [inline phalcon/image/adapter/imagick.zep]: for what is this preg_match ? Just compare below against 7.0.0

**Comment by Izopi4a** [comment]
i will fix windows build these days because i need to dig into that

**Comment by Jurigag** [comment]
`Class '\Imagick' does not implement static method: 'getVersion'`, hmmm imagick not loaded on windows or what ?

**Comment by sergeyklay** [comment]
Fixed in the `3.0.x` branch.

**Pull request closed without merging**